code/datums/status_effects/buffs.dm

```diff
@@ -72,16 +76,18 @@
 /datum/status_effect/vanguard_shield/on_apply()
 	owner.log_message("gained Vanguard stun immunity", LOG_ATTACK)
 	owner.add_stun_absorption("vanguard", INFINITY, 1, "'s yellow aura momentarily intensifies!", "Your ward absorbs the stun!", " radiating with a soft yellow light!")
-	owner.visible_message("<span class='warning'>[owner] begins to faintly glow!</span>", "<span class='brass'>You will absorb all stuns for the next twenty seconds.</span>")
+	owner.visible_message("<span class='warning'>[owner] begins to faintly glow!</span>", "<span class='brass'>You will absorb all stuns aswell as quickly regenerate stamina for the next twenty seconds .</span>")
 	owner.SetAllImmobility(0, FALSE)
-	owner.setStaminaLoss(0, FALSE)
 	progbar = new(owner, duration, owner)
 	progbar.bar.color = list("#FAE48C", "#FAE48C", "#FAE48C", rgb(0,0,0))
 	progbar.update(duration - world.time)
 	return ..()
 
 /datum/status_effect/vanguard_shield/tick()
 	progbar.update(duration - world.time)
+	var/oldstamloss = owner.getStaminaLoss()
+	owner.adjustStaminaLoss(-6) //up to 30 stam / second for now, lets see...
```

Comment:
this is 3/second

Reply:
Runs on status effect ticks (0.2 seconds I believe), not life ticks, it just calls a value adjustment proc.